v2/tools/generator/internal/config/object_model_configuration_test.go

```diff
@@ -43,6 +43,19 @@ func TestObjectModelConfiguration_WhenYAMLBadlyFormed_ReturnsError(t *testing.T)
 	g.Expect(err).NotTo(Succeed())
 }
 
+func TestObjectModelConfiguration_WhenYAMLHasDuplicateGroups_ReturnsError(t *testing.T) {
+	t.Parallel()
+	g := NewGomegaWithT(t)
+
+	yamlBytes := loadTestData(t)
+
+	var model ObjectModelConfiguration
+	err := yaml.Unmarshal(yamlBytes, &model)
+	g.Expect(err).NotTo(Succeed())
+	g.Expect(err.Error()).To(ContainSubstring("duplicate group"))
+	g.Expect(err.Error()).To(ContainSubstring("microsoft.person"))
```

Comment:
Again, prefer use of `MatchError()`. 

Also, please check all your tests for this pattern and update them too.